### PR TITLE
Travis integrated

### PR DESCRIPTION
**Pull request by stephenrlouie**
- ~Currently failing builds because docker tests mounting issue~
- ~Currently pushing containers to here for successful builds. will move to a sigmadev account once we have a dockerhub account active.~

# Good to merge
- Pushes containers to [sigmadev](https://hub.docker.com/u/sigmadev/) .
    - we will push all successful builds to [sigma-dev/sigma-dev](https://hub.docker.com/r/sigmadev/sigma-dev/tags/) with the tags being the commit sha.
    - merges to master will go to [sigma-dev/sigma](https://hub.docker.com/r/sigmadev/sigma/tags/) with the tag latest. *This won't exist until this hits master.*
    - tags will also be pushed to  [sigma-dev/sigma](https://hub.docker.com/r/sigmadev/sigma/tags/), with the appropriate tags

- All builds are posted in [slack sigma-dev#builds](https://sigma-dev.slack.com/messages/C53PQRS80) *I can just push master builds here, given our size I figured all builds worked*

[Sample dockerhub](https://hub.docker.com/u/stephenrlouie/) Look at the `travis-dev` and `travis-prod` repos. [Source](https://github.com/stephenrlouie/travis)

### CODE REVIEW

**Comment by adamschaub** [comment]
Looks good. Lots simpler than Jenkins

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - COMMIT=${TRAVIS_COMMIT::8}
+    - DEV_REPO=sigmadev/sigma-dev
+    - PROD_REPO=sigmadev/sigma
+script:
+  - make test
+  - make
+  - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
+  - docker build -f Dockerfile -t $DEV_REPO:$COMMIT .
+  - docker push $DEV_REPO
+  - if [ "$TRAVIS_BRANCH" = "master" -a "$TRAVIS_PULL_REQUEST" = "false" ]; then docker tag $DEV_REPO:$COMMIT $PROD_REPO:latest && docker push $PROD_REPO; fi
```
Comment (adamschaub) [inline .travis.yml]: Possible to make these multi line? Just for the sake of readability

Comment (stephenrlouie) [inline .travis.yml]: done

**Comment by stephenrlouie** [comment]
I was hoping to ENV var the slack stuff, but they decided not to take this feature on. https://github.com/travis-ci/travis-ci/issues/6387

The encrypted token is good enough.

**Pull request merged**